src/chain/member_info.rs

```diff
@@ -7,18 +7,40 @@
 // permissions and limitations relating to use of the SAFE Network Software.
 
 /// Information about a member of our section.
+
+pub type AgeCounter = u32;
+
+/// The minimum allowed value of the Age Counter
+pub const MIN_AGE: AgeCounter = 16;
+
+const MAX_INFANT_AGE: u8 = 4;
+
 #[derive(Clone, Copy, Eq, PartialEq, Hash, Serialize, Deserialize, Debug)]
 pub struct MemberInfo {
     pub persona: MemberPersona,
-    pub age: u8,
+    pub age_counter: AgeCounter,
     pub state: MemberState,
 }
 
+impl MemberInfo {
+    pub fn age(self) -> u8 {
+        f64::from(self.age_counter).log2() as u8
+    }
+
+    #[allow(unused)]
+    pub fn increase_age(&mut self) {
+        self.age_counter += 1;
+        if self.age() > MAX_INFANT_AGE && self.persona == MemberPersona::Infant {
+            self.persona = MemberPersona::Adult;
```

Comment:
Do we even need the `persona` field? It looks like it can be fully determined from other pieces of info:
```
if node is in elders_info => Elder
else if age > MAX_INFANT_AGE => Adult
else => Infant
```

Reply:
Hmm, that's true. I'd probably keep the `MemberPersona` type, though, should still be useful in some places.